src/rcv/septentrio.c

```diff
@@ -99,824 +217,1580 @@ static uint8_t sig_tbl[SBF_MAXSIG+1][2]={ /* system, obs-code */
+ *          -JL1L   : select 1L for QZS L1 (default: 1C)
+ *          -JL1Z   : select 1Z for QZS L1 (default: 1C)
+ *          -CL1P   : select 1P for BDS B1 (default: 2I)
+ *          -GALINAV: select I/NAV for Galileo ephemeris (default: all)
+ *          -GALFNAV: select F/NAV for Galileo ephemeris (default: all)
+ *-----------------------------------------------------------------------------*/
+extern int input_sbf(raw_t *raw, uint8_t data) {
+  trace(5, "input_sbf: data=%02x\n", data);
+
+  if (raw->nbyte == 0) {
+    if (sync_sbf(raw->buff, data))
+      raw->nbyte = 2;
+    return 0;
+  }
+  raw->buff[raw->nbyte++] = data;
+  if (raw->nbyte < 8)
+    return 0;
+
+  if ((raw->len = U2(raw->buff + 6)) > MAXRAWLEN) {
+    trace(2, "sbf length error: len=%d\n", raw->len);
+    raw->nbyte = 0;
+    return -1;
+  }
+  if (raw->nbyte < raw->len)
+    return 0;
+  raw->nbyte = 0;
+
+  /* decode SBF block */
+  return decode_sbf(raw);
 }
 /* input SBF raw data from file ------------------------------------------------
-* fetch next SBF raw data and input a message from file
-* args   : raw_t *raw       IO  receiver raw data control struct
-*          FILE  *fp        I   file pointer
-* return : status(-2: end of file, -1...9: same as above)
-*-----------------------------------------------------------------------------*/
-extern int input_sbff(raw_t *raw, FILE *fp)
-{
-    int i,data;
-    
-    trace(4,"input_sbff:\n");
-    
-    if (raw->nbyte==0) {
-        for (i=0;;i++) {
-            if ((data=fgetc(fp))==EOF) return -2;
-            if (sync_sbf(raw->buff,(uint8_t)data)) break;
-            if (i>=4096) return 0;
-        }
-    }
-    if (fread(raw->buff+2,1,6,fp)<6) return -2;
-    raw->nbyte=8;
-    
-    if ((raw->len=U2(raw->buff+6))>MAXRAWLEN) {
-        trace(2,"sbf length error: len=%d\n",raw->len);
-        raw->nbyte=0;
-        return -1;
+ * fetch next SBF raw data and input a message from file
+ * args   : raw_t *raw       IO  receiver raw data control struct
+ *          FILE  *fp        I   file pointer
+ * return : status(-2: end of file, -1...9: same as above)
+ *-----------------------------------------------------------------------------*/
+extern int input_sbff(raw_t *raw, FILE *fp) {
+  int i, data;
+  int lenToRead;
+
+  trace(4, "input_sbff:\n");
+
+  if (raw->nbyte == 0) {
+    for (i = 0;; i++) {
+      if ((data = fgetc(fp)) == EOF)
+        return -2;
+      if (sync_sbf(raw->buff, (uint8_t)data))
+        break;
+      if (i >= 4096)
+        return 0;
     }
-    if (raw->len!=0&&fread(raw->buff+8,raw->len-8,1,fp)<1) return -2;
-    raw->nbyte=0;
-    
-    /* decode SBF block */
-    return decode_sbf(raw);
-}
+  }
+  if (fread(raw->buff + 2, 1, 6, fp) < 6) {
+    trace(2, "sbf fread failed\n");
+    return -2;
+  }
+  raw->nbyte = 8;
 
+  if ((raw->len = U2(raw->buff + 6)) > MAXRAWLEN) {
+    trace(2, "sbf length error: len=%d\n", raw->len);
+    raw->nbyte = 0;
+    return -1;
+  }
+  if (raw->len <= 8) {
```

Comment:
This <8 catch was the "bugfix" for broken files.

In the testing file, it had a malformed message at the start of the file.
This would cause the length calc of raw->len-8 to go negative
Which then would cause fread to throw an error.
This then aborted the conversion.

So we instead treat these as skippable events and it will resync.

Reply:
lmao wtf

Reply:
Yep, makes sense. I do recall looking seeing similar issues elsewhere that I've handled in the Uploader by default. Good catch ✋

Reply:
I just happened to walk into the perfect "everything is on fire" file where the first ~ hundred bytes are borked :)